client/web/src/open-in-editor/build-url.ts

```diff
@@ -0,0 +1,142 @@
+import * as path from 'path'
+
+import type { UIRangeSpec } from '@sourcegraph/shared/out/src/util/url'
+
+import type { EditorReplacements, EditorSettings } from './editor-settings'
+import { Editor, getEditor, supportedEditors } from './editors'
+
+export function buildEditorUrl(
+    repoBaseNameAndPath: string,
+    range: UIRangeSpec['range'] | undefined,
+    editorSettings: EditorSettings | undefined,
+    sourcegraphBaseUrl: string
+): URL {
+    const editorSettingsErrorMessage = getEditorSettingsErrorMessage(editorSettings, sourcegraphBaseUrl)
+    if (editorSettingsErrorMessage) {
+        throw new TypeError(editorSettingsErrorMessage)
+    }
+
+    const projectPath = getProjectPath(editorSettings || {}) as string // Default is only there to soothe TypeScript
+    const editor = getEditor(editorSettings?.editorId ?? 'vscode') as Editor // Default is only there to soothe TypeScript
+    const urlPattern = getUrlPattern(editor, editorSettings || {}) // Default is only there to soothe TypeScript
+    // If VS Code && (Windows || UNC flag is on), add an extra slash in the beginning
+    const pathPrefix =
+        editor.id === 'vscode' && (isWindowsPath(projectPath) || editorSettings?.['vscode.isProjectPathUNCPath'])
+            ? '/'
+            : ''
+
+    const absolutePath = path.join(projectPath, repoBaseNameAndPath)
+    const { line, column } = range ? { line: range.start.line, column: range.start.character } : { line: 1, column: 1 }
+    const url = urlPattern
+        .replace('%file', pathPrefix + absolutePath)
+        .replace('%line', `${line}`)
+        .replace('%col', `${column}`)
+    return new URL(doReplacements(url, editorSettings?.replacements))
+}
+
+export function getEditorSettingsErrorMessage(
+    editorSettings: EditorSettings | undefined,
+    sourcegraphBaseUrl: string
+): string | undefined {
+    const learnMoreURL = new URL('https://docs.sourcegraph.com/integration/open_in_editor').href
```

Comment:
Isn't this equal to just doing:

```suggestion
    const learnMoreURL = 'https://docs.sourcegraph.com/integration/open_in_editor'
```